packages/build/src/status/validations.ts

```diff
@@ -0,0 +1,24 @@
+import { DynamicMethods } from 'packages/js-client/lib/types.js'
+
+import { SecretScanResult } from '../plugins_core/secrets_scanning/utils.js'
+import { SystemLogger } from '../plugins_core/types.js'
+
+// Reports any validations completed on the deploy to the API
+export const reportValidations = async function ({
+  api,
+  secretScanResult,
+  deployId,
+  systemLog,
+}: {
+  api: DynamicMethods
+  secretScanResult: SecretScanResult
+  deployId: string
+  systemLog: SystemLogger
+}) {
+  try {
+    // @ts-expect-error Property 'updateDeployValidations' does not exist on type 'DynamicMethods'. This is a private/internal-only method and isn't generated in the type definitions.
+    await api.updateDeployValidations({ deploy_id: deployId, body: { secrets_scan: secretScanResult } })
+  } catch (e) {
```

Comment:
Ideally we are catching specific named errors here, but it's possible we don't know at this time what could go wrong with this call.

Reply:
a malformed payload is what springs to mind, but it's still behind a feature flag and being iterated on, so I'd suggest we monitor initially and improve as we go